Sources/TuistKit/Services/EditService.swift

```diff
@@ -56,27 +58,22 @@ final class EditService {
         let plugins = await loadPlugins(at: path)
 
         if !permanent {
-            try withTemporaryDirectory(removeTreeOnDeinit: true) { generationDirectory in
-                EditService.temporaryDirectory = generationDirectory
-
-                signalHandler.trap { _ in
-                    try? EditService.temporaryDirectory.map(FileHandler.shared.delete)
-                    exit(0)
-                }
-
-                guard let selectedXcode = try XcodeController.shared.selected() else {
-                    throw EditServiceError.xcodeNotSelected
-                }
-
-                let workspacePath = try projectEditor.edit(
-                    at: path,
-                    in: generationDirectory,
-                    onlyCurrentDirectory: onlyCurrentDirectory,
-                    plugins: plugins
-                )
-                logger.pretty("Opening Xcode to edit the project. Press \(.keystroke("CTRL + C")) once you are done editing")
-                try opener.open(path: workspacePath, application: selectedXcode.path, wait: true)
+            let pathHash = path.hashValue
+            let cacheDirectory = try cacheDirectoryProviderFactory.cacheDirectories()
+            let generationDirectory = try cacheDirectory.tuistCacheDirectory(for: .generatedEditProjects).appending(component: "\(pathHash)")
+            
+            guard let selectedXcode = try XcodeController.shared.selected() else {
+                throw EditServiceError.xcodeNotSelected
             }
```

Comment:
What if we add a new variant of the `selected` method that throws instead of doing that here? I think this might be useful in other places, too

Reply:
I'm not sure I understand this one. The selected() method already throws. Should I just remove the guard and let it throw?

Reply:
Or, do you mean have a new variant that throws, but also returns a non-optional Xcode?

Reply:
Alright, not sure what naming you were looking for, but I added a new method to XcodeController called 

`func guaranteed() throws -> Xcode`

Reply:
I've looked at the usages of `selected()` and in _every_ case, we `guard let` it. I think we can change the `selected()` variant to always throw if the Xcode version is not found (and remove the `guaranteed` function completely as it should be the new "default")

Reply:
Alright, but should the existing call sites surface the noXcode error, or should they continue to return nil (like in ProjectEditorMapper.swift:249?

Reply:
I would lean to throwing an error, I don't really see in which scenario `xcode-select` wouldn't work. `selectedVersion()` already throws an error actually

Reply:
Alright, that's what I did, thanks